iocage/Jail.py

```diff
@@ -1683,34 +1678,30 @@ def _launch_parameters(self) -> typing.Dict[str, typing.Union[str, int]]:
         p["persist"] = 1
         return p
 
-    def _launch_persistent_jail(
-        self,
-        passthru: bool
-    ) -> iocage.helpers.CommandOutput:
-
-        command = self._launch_command + [
-            "persist",
-            f"exec.poststart=\"{self.get_hook_script_path('poststart')}\""
-        ]
+    def _launch_jail(self) -> jail.RunningJail:
+        stopped_jail = jail.StoppedJail(str(self.root_dataset.mountpoint))
+        stopped_jail.parameters = self._launch_parameters
 
-        stdout, stderr, returncode = self._exec_host_command(
-            command=command,
-            passthru=passthru,
-            env=self.env
-        )
-        if returncode > 0:
+        try:
+            return jail.start()
+        except Exception as e:
             self.logger.verbose(
                 f"Jail '{self.humanreadable_name}' was not started"
             )
-            return stdout, stderr, returncode
+            raise e
+
+    def _launch_persistent_jail(
+        self,
+        passthru: bool
+    ) -> None:
+
+        running_jail = self._launch_jail()
```

Comment:
It might make more sense to do this the other way around, and call `self._launch_persistent_jail()` from `_launch_jail()`, as `libjail-rs` jails are by default persistent, and you can drop persistence after spawning the command using `jail.RunningJail.defer_cleanup`.